library/common/jni/BUILD

```diff
@@ -17,27 +17,64 @@ envoy_cc_library(
     ],
     repository = "@envoy",
     deps = [
+        "//:jni_headers",
         "//library/common/types:c_types_lib",
         "@envoy//source/common/common:assert_lib",
     ],
 )
 
+# Produces two binaries: one for Android, and one for tests.
```

Comment:
I was able tohandle this bazel gymnastics for you now!

https://github.com/envoyproxy/envoy-mobile/blob/main/library/common/jni/BUILD#L47-L71 Is exactly what you're looking for. You can rebase your fork and branch from envoy-mobile (if you're having trouble, I can help you too)

Reply:
This I fear is not helping moving forward with Cronvoy. I can't use the "java_jni_lib" BUILD target to construct the Cronvoy tests. The Cronvoy tests depends on the libenvoy_jni.so BUILD target.